neurom/fst/_io.py

```diff
@@ -111,6 +111,30 @@ def load_neuron(filename):
 update_wrapper(load_neurons, _iout.load_neurons)
 
 
+def _merge_sections(sec_a, sec_b):
+    '''Merge two sections
+
+    Merges sec_a into sec_b and sets sec_b attributes to default
+    '''
+    sec_b.ids = sec_a.ids + sec_b.ids[1:]
+    sec_b.ntype = sec_a.ntype
+    sec_b.pid = sec_a.pid
+    sec_a.ids = []
+    sec_a.pid = -1
+
+
+def _section_end_points(data_block):
+    '''Get the section end-points '''
+    # number of children per point
+    n_children = defaultdict(int)
+    for row in data_block:
+        n_children[int(row[COLS.P])] += 1
+
+    # end points have either no children or more than one
+    return set(i for i, row in enumerate(data_block)
+               if n_children[row[COLS.ID]] != 1)
+
+
 def extract_sections(data_block):
```

Comment:
This function is huge and awful. I need to find a better way to do this in the future.